--!strict

------------------------------------------------------------------------------------------------------------------------

--[[

	Junction - Manages functions used for events under a single connection.
	By @Cayla

]]

------------------------------------------------------------------------------------------------------------------------

local Junction = {
	Pools = {},
}

------------------------------------------------------------------------------------------------------------------------

local Pool = require("@self/Pool")

------------------------------------------------------------------------------------------------------------------------

function Junction.GetPool(Event: RBXScriptSignal): Pool.Pool
	if not Junction.Pools[Event] then
		Junction.Pools[Event] = Pool.new(Event)
	end

	return Junction.Pools[Event]
end

------------------------------------------------------------------------------------------------------------------------

return Junction
